numba/tests/test_dispatcher.py

```diff
@@ -1021,6 +1022,73 @@ def execute_with_input():
         err = execute_with_input()
         self.assertEqual(err.strip(), "cache hits = 1")
 
+    def test_user_set_cpu_name(self):
+        self.check_pycache(0)
+        mod = self.import_module()
+        mod.self_test()
+        cache_size = len(self.cache_contents())
+
+        mtimes = self.get_cache_mtimes()
+        # Change CPU name to generic
+        try:
+            os.environ['NUMBA_CPU_NAME'] = 'generic'
```

Comment:
This is not an issue for the current `multiprocessing.Pool` behavior.  Parallel tests are running in different processes and there are no context-switching (a pool-worker does not switch to another item unless it is idle).  However, it is safer to make sure that these tests are running in serial to avoid relying on specific behavior in the Pool implementation.